ckanext/switzerland/helpers/date_helpers.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+import logging
+from dateutil.parser import parse
+import ckan.plugins.toolkit as tk
+
+DATE_FORMAT = tk.config.get(
+    'ckanext.switzerland.date_picker_format', '%d.%m.%Y')
+
+log = logging.getLogger(__name__)
+
+
+def isodate_or_none_for_storage(value):
+    try:
+        value_as_isoformat = parse(value).isoformat()
+        if value_as_isoformat == value:
+            return value
+    except Exception:
+        return None
+
+
+def ogdch_date_or_none_for_storage(value):
+    try:
+        return datetime.strptime(value, DATE_FORMAT).isoformat()
+    except Exception:
+        return None
+
+
+def timestamp_date_or_none_for_storage(value):
+    try:
+        return datetime.fromtimestamp(int(value)).isoformat()
+    except Exception:
+        return None
+
+
+def datetime_or_none_for_storage(value):
+    try:
+        if isinstance(value, datetime):
+            return value.isoformat()
+    except Exception:
+        return None
+
+
+def isodate_or_none_for_display(value):
+    try:
+        value_datetime = parse(value)
+        if value_datetime.isoformat() == value:
+            return get_ogdch_date(value_datetime)
+    except Exception:
+        return None
+
+
+def ogdch_date_or_none_for_display(value):
+    try:
+        if datetime.strptime(value, DATE_FORMAT):
```

Comment:
Question for this function and the next two:
should we check here if date format is isoformat and then display?
`if datetime.strptime(value, DATE_FORMAT).isoformat():`